### PR TITLE
Fix QUnit tests in iOS browsers (close #113)

### LINKED ISSUE #113: Fix QUnit tests in iOS browsers (Safari, iPhone, iPad)
It seems, problems are the same in all the browsers

https://saucelabs.com/tests/8679452a63474586af2d33eed02a589e
https://saucelabs.com/tests/e7498fd2bde04c80b7e77be44254436a
https://saucelabs.com/tests/5213d7175e8345bc8424ebd2a8e3d5c3

### PR DESCRIPTION
**Pull request by helen-dikareva**
/cc @inikulin @AndreyBelym

### CODE REVIEW

File: src/client/runner/automation/playback/click.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ function clickOnSelectChildElement (childElement, clickOptions, actionCallback,
             },
 
             click: function () {
-                var clickLeadChanges = isClickOnOption && !targetElement.disabled;
+                var clickLeadChanges = isClickOnOption && !targetElement.disabled && childIndex !== selectedIndex;
```
Comment (inikulin) [inline src/client/runner/automation/playback/click.js]: `clickCausesChange`?

File: src/client/runner/automation/playback/key-press-simulator.js [hunk 2]
```diff
@@ -580,20 +580,28 @@ var supportedShortcutHandlers = (function () {
...
+                    if (browserUtils.isSafari)
+                        textInputTagNameRegExp = /^(text|password|color|date|time|datetime|datetime-local|email|month|number|search|tel|url|week|image)$/i;
+                    else if (browserUtils.isFirefox)
+                        textInputTagNameRegExp = /^(text|password|date|time|datetime|datetime-local|email|month|number|search|tel|url|week|image)$/i;
+                    else if (browserUtils.isIE)
+                        textInputTagNameRegExp = /^(text|password|color|date|time|datetime|datetime-local|email|file|month|number|search|tel|url|week|image)$/i;
+                    else
+                        textInputTagNameRegExp = /^(text|password|datetime|email|number|search|tel|url|image)$/i;
+
+                    if (textInputTagNameRegExp.test(element.type)) {
```
Comment (inikulin) [inline src/client/runner/automation/playback/key-press-simulator.js]: This definitely should be refactored to the utility function

**Comment by inikulin** [comment]
\r-

**Comment by helen-dikareva** [comment]
FPR

File: src/client/runner/automation/playback/key-press-simulator.js [hunk 3]
```diff
@@ -580,20 +593,19 @@ var supportedShortcutHandlers = (function () {
                 else {
                     //the form is also submitted on enter press if there is only one input of the following types on it
                     //  and this input is focused (http://www.w3.org/TR/html5/forms.html#implicit-submission)
-                    var textInputTypeRegExp = browserUtils.isWebKit ? /^(text|search|url|number|tel|password|number|email)$/i :
-                                              /^(text|search|url|number|tel|password|number|email|date|time)$/i;
+                    var textInputTagNameRegExp = getTextInputTagNameRegExp();
```
Comment (inikulin) [inline src/client/runner/automation/playback/key-press-simulator.js]: Why not call it `isTextInput`, move re test logic there and move it to the DOM utils?

**Comment by inikulin** [comment]
\r-

File: src/client/runner/automation/playback/key-press-simulator.js (hunk 3 above)
Comment (helen-dikareva) [inline src/client/runner/automation/playback/key-press-simulator.js]: I don't like 'is TextInput' title. This method's used only for this scenario for form and doesn't correspond only inputs with 'type=text' (or inputs look like input 'type=text'). I think we can use just
```var inputTypeRegExp = getInputTypeRegExp();```

Comment (VasilyStrelyaev) [inline src/client/runner/automation/playback/key-press-simulator.js]: standard calls such fields *fields that blocks implicit submission*
so, maybe `blocksImplicitSubmission(el)`

Comment (VasilyStrelyaev) [inline src/client/runner/automation/playback/key-press-simulator.js]: this is a weird name, but it's standard - I'm not sure we need to coin a new one

**Comment by inikulin** [comment]
Doesn't seems like an accidental fail. On test doesn't work in iOS browsers

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
Argh, tests fails only with @testcafe-build-bot When I try to reproduce this with manual session on SauceLabs thay absolutely stable.
Don't look at this I'll do disgusting things to stable it)
Wait for my FPR comment

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
FPR

**Comment by helen-dikareva** [comment]
Sorry, I forgot about approved by Vasily comment. Now it's FPR

File: src/client/runner/automation/playback/key-press-simulator.js [hunk 4]
```diff
@@ -578,22 +578,20 @@ var supportedShortcutHandlers = (function () {
                 if (submitButton)
                     eventSimulator.click(submitButton);
                 else {
-                    //the form is also submitted on enter press if there is only one input of the following types on it
-                    //  and this input is focused (http://www.w3.org/TR/html5/forms.html#implicit-submission)
-                    var textInputTypeRegExp = browserUtils.isWebKit ? /^(text|search|url|number|tel|password|number|email)$/i :
-                                              /^(text|search|url|number|tel|password|number|email|date|time)$/i;
-
-                    if (textInputTypeRegExp.test(element.type)) {
+                    // NOTE: the form is also submitted on enter key press if there is only one input of certain
```
Comment (AlexanderMoskovkin) [inline src/client/runner/automation/playback/key-press-simulator.js]: Let's move the comment to the function definition place

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
FPR again

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by inikulin** [comment]
lgtm

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Pull request merged**